x-pack/plugins/apm/public/components/app/Home/index.tsx

```diff
@@ -60,7 +61,7 @@ export function Home({ tab }: Props) {
   ) as $ElementType<typeof homeTabs, number>;
 
   return (
-    <>
+    <EnvironmentsContextProvider>
```

Comment:
Any reason not to add this context at the very top-level since it's needed everywhere (afaict)?
https://github.com/elastic/kibana/blob/37cc7af49d7f1f81f8e6265d05dc1fd355b0ca24/x-pack/plugins/apm/public/application/index.tsx#L80

Reply:
React Router's `useParams` hook only returns route parameters for the Routes that are rendering the current component. As `serviceName` is a route parameter of the service routes, it's not available when the EnvironmentsContextProvider is rendered top-level.

Reply:
Okay, makes sense. I recall having the same issue with `ApmServiceContextProvider` and adding it on the route level instead:

https://github.com/elastic/kibana/blob/f7fdda5db7e41b04f0f37cc180b441e8ed0c9321/x-pack/plugins/apm/public/components/app/Main/route_config/index.tsx#L312

Would it make sense to handle `EnvironmentsContextProvider` the same way?

Reply:
I think I'm already doing it that way? At least,I'm piggy-backing on `withApmServiceContext` in route_config/index.tsx. Or do you mean something different?